src/lib/syncable_ledger/syncable_ledger.ml

```diff
@@ -614,9 +507,14 @@ end = struct
                 "Peer $peer answered question we didn't ask! Query was $query \
                  answer was $answer"
         in
-        if Valid.completely_fresh t.validity then (
+        if
+          (not (MT.has_unset_slots t.tree))
```

Comment:
I don't think we need `hash_unset_slots` because almost all of the ledgers has this value set to false except for the "regular" `ledger`. We are not using that anymore. Can you remove this?

Reply:
The reason it's in there is that the tests fail if you don't do it. They use `Merkle_ledger.Ledger`.